src/tests/Microsoft.PowerFx.Interpreter.Tests/CustomFunctions.cs

```diff
@@ -3,17 +3,132 @@
 
 using System;
 using System.Linq;
+using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.PowerFx.Core.IR;
 using Microsoft.PowerFx.Core.Tests;
+using Microsoft.PowerFx.Functions;
+using Microsoft.PowerFx.Interpreter;
 using Microsoft.PowerFx.Types;
 using Xunit;
 
 namespace Microsoft.PowerFx.Tests
 {
     public class CustomFunctions : PowerFxTest
     {
+        public class CustomFunctionError : ReflectionFunction
+        {
+            public CustomFunctionError()
+                : base("CustomFunctionError", FormulaType.Number, FormulaType.Number) 
+            {
+            }
+
+            public static NumberValue Execute(NumberValue arg) 
+            {
+                if (arg.Value < 0)
+                {
+                    throw new CustomFunctionErrorException("arg should be greater than 0");
```

Comment:
>CustomFunctionErrorException [](http://example.com/codeflow?start=30&length=28)

Great that you're testing both paths. 
nit - we should add some comments since people will use these tests as documentation. 

Line 31:
// CustomFunctionErrorException is caught and converted to error value 

Line 35:
// All other exceptions are uncaught and will abort the eval.